### PR TITLE
Discovery: Add filters to allow disabling metadata output

### LINKED ISSUE #4913: Jetpack: Metadata Duplication
## Feature Description

This is a follow-up to #3092

While that one works great, it results in duplicate Open Graph tags being printed because Jetpack does not remove existing ones.

Seems like they'd need to do this:

```php
$instance = Google\Web_Stories\get_plugin_instance()->discovery;
\remove_action( 'web_stories_story_head', [ $instance, 'print_open_graph_metadata' ] );
\remove_action( 'web_stories_story_head', [ $instance, 'print_twitter_metadata' ] );
```

Not sure if they're open to do this.

Alternatively we remove those two actions if we detect Jetpack.

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Add a simple filter to enable / disable open graph tags. 

## Testing Instructions

---

Fixes #4913

### CODE REVIEW

File: includes/Discovery.php [hunk 1]
```diff
@@ -47,8 +47,19 @@ class Discovery {
...
-		add_action( 'web_stories_story_head', [ $this, 'print_open_graph_metadata' ] );
-		add_action( 'web_stories_story_head', [ $this, 'print_twitter_metadata' ] );
+
+		/**
+		 * Filters filter to enable / disable open graph data.
+		 *
+		 * @since 1.2.0
+		 * @param bool $enable_open_graph Enable / disable open graph. Default to true.
+		 */
+		$enable_open_graph = apply_filters( 'web_stories_enable_open_graph', true );
```
Comment (swissspidy) [inline includes/Discovery.php]: After looking at this, I think we need more fine-grained filters for each of these methods.

Plugins like Yoast SEO need to disable all of them, while Jetpack only would need to disable Open Graph & Twitter output.

So there could be a few filters:

* `web_stories_print_metadata`
* `web_stories_print_schemaorg_metadata`
* `web_stories_print_open_graph_metadata`
* `web_stories_print_twitter_metadata`

Comment (spacedmonkey) [reply includes/Discovery.php]: @swissspidy Is this what you had in mind - https://github.com/google/web-stories-wp/pull/5312/commits/2cc1e9e3296dc237632d487f71f67d360e681f1b ?

Comment (swissspidy) [reply includes/Discovery.php]: Yep!

Comment (spacedmonkey) [reply includes/Discovery.php]: Fixed.

File: includes/Discovery.php [hunk 2]
```diff
@@ -47,6 +47,7 @@ class Discovery {
 	public function init() {
 		add_action( 'web_stories_story_head', [ $this, 'print_metadata' ] );
 		add_action( 'web_stories_story_head', [ $this, 'print_schemaorg_metadata' ] );
+
```
Comment (swissspidy) [inline includes/Discovery.php]:
```suggestion
```

**Review by swissspidy — APPROVED** [review]

**Pull request merged**